### PR TITLE
Hashtag-Widget (IOS-152)

### PR DESCRIPTION
**Pull request by zeitschlag**
- Adds a Widget to show the test of a status, that contains a given Hashtag.
- Hashtag can be defined in the Widget-Settings

<div>

</div>

### CODE REVIEW

File: WidgetExtension/Variants/Hashtag/HashtagWidgetView.swift [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        }
+    }
+}
+
+/// Inspired by: https://swiftuirecipes.com/blog/swiftui-text-with-html-via-nsattributedstring
+extension Text {
+    init(statusHTML htmlString: String, fontSize: Int = 16, fontWeight: Int = 400, colorScheme: ColorScheme = .light) {
+
+        let textColor = (colorScheme == .light) ? "#000000" : "#FFFFFF"
+        let accentColor = (colorScheme == .light) ? "#563ACC" : "#858AFA"
```
Comment (DK1DA) [inline WidgetExtension/Variants/Hashtag/HashtagWidgetView.swift]: Just a long shot and an idea, no need to take this, but maybe we should convert the `UIColor` from our Asset catalogs to hex here e.g. using:

```swift
extension UIColor {
  var hexValue: String {
    let components = cgColor.components
      
    let red: CGFloat = components?[0] ?? 0.0
    let green: CGFloat = components?[1] ?? 0.0
    let blue: CGFloat = components?[2] ?? 0.0

    return String(
        format: "#%02lX%02lX%02lX",
        lroundf(Float(red * 255)),
        lroundf(Float(green * 255)),
        lroundf(Float(blue * 255))
    )
  }
}
```

**Review by kimar — APPROVED** [review]
LGTM 👏

File: WidgetExtension/Variants/Hashtag/HashtagWidgetView.swift (hunk 1 above)
Comment (j-f1) [reply WidgetExtension/Variants/Hashtag/HashtagWidgetView.swift]: Or: `return "rgb(\(red * 255) \(green * 255) \(blue * 255))"`

File: WidgetExtension/Variants/Hashtag/HashtagWidgetView.swift [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+"""
+
+        let attributedString: NSAttributedString
+        if let data = fullHTML.data(using: .unicode),
+           let attrString = try? NSAttributedString(data: data,
+                                                    options: [.documentType: NSAttributedString.DocumentType.html],
+                                                    documentAttributes: nil) {
+            attributedString = attrString
+        } else {
+            attributedString = NSAttributedString()
```
Comment (j-f1) [inline WidgetExtension/Variants/Hashtag/HashtagWidgetView.swift]: Not great UX but better than blank IMO

```suggestion
            attributedString = NSAttributedString(string: htmlString)
```

File: Localization/app.json [hunk 3]
```diff
@@ -870,6 +870,22 @@
             "configuration_description": "Show latest followers.",
             "title": "Latest followers",
             "last_update": "Last update: %s"
+        },
+        "hashtag": {
+            "configuration": {
+                "display_name": "Hashtag",
+                "description": "Shows a recent status with the selected hashtag."
```
Comment (j-f1) [inline Localization/app.json]:
```suggestion
                "description": "Shows a recent post with the selected hashtag."
```

File: Localization/app.json [hunk 4]
```diff
@@ -870,6 +870,22 @@
...
+            },
+            "not_found": {
+                "account_name": "John Mastodon",
+                "account": "@johnMastodon@no-such.account",
+                "content": "Sorry, we didn't find a status with <a>#%@</a>. Please try a <a>#DifferentHashtag</a> or check the Widget-settings."
```
Comment (j-f1) [inline Localization/app.json]:
```suggestion
                "content": "Sorry, we couldn’t find any posts with the hashtag <a>#%@</a>. Please try a <a>#DifferentHashtag</a> or check the widget settings."
```

File: Localization/app.json [hunk 5]
```diff
@@ -870,6 +870,22 @@
...
+            },
+            "placeholder": {
+                "account_name": "John Mastodon",
+                "account": "@johnMastodon@no-such.account",
+                "content": "This is how a status with a <a>#hashtag</a> would look like. Of course you can decide about the <a>#hashtag</a>"
```
Comment (j-f1) [inline Localization/app.json]:
```suggestion
                "content": "This is how a post with a <a>#hashtag</a> would look. Pick whichever <a>#hashtag</a> you want in the widget settings."
```

File: WidgetExtension/Variants/Hashtag/HashtagWidgetView.swift (hunk 1 above)
Comment (zeitschlag) [reply WidgetExtension/Variants/Hashtag/HashtagWidgetView.swift]: It's a good idea, so: Thank you both!

**Pull request merged**